domain/src/main/java/org/oppia/android/domain/exploration/ExplorationSessionTimerController.kt

```diff
@@ -0,0 +1,59 @@
+package org.oppia.android.domain.exploration
+
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.domain.oppialogger.analytics.ApplicationLifecycleListener
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * Controller to decide when the session timer is started and stopped based on signals received
+ * from [ExplorationProgressListener] and [ApplicationLifecycleListener].
+ */
+@Singleton
+class ExplorationSessionTimerController @Inject constructor(
+  private val explorationActiveTimeController: ExplorationActiveTimeController
+) : ExplorationProgressListener, ApplicationLifecycleListener {
+  private var isAppInForeground: Boolean = false
+  private var explorationStarted: Boolean = false
+  private var profileId: ProfileId? = null
+  private var topicId: String? = null
```

Comment:
It seems like these variables will be accessed on different threads. These may need to be transferred over to an internal data class combined with either a command pattern (like we use in ``ExplorationProgressController`` via an actor) or to ``InMemoryBlockingCache`` to ensure synchronous access to the data).

Reply:
I chose to use the command pattern approach since I could not find sufficient references to InMemoryBlockingCache to learn how to use it, due to limited use in the codebase.

This refactor, though it ended up being non-trivial, allowed me to have better control of the timer, since the timer now has its own session that I can start, pause, resume and stop.

Reply:
The logic for this seems to have been moved to the `ExplorationActiveTimeController` class which is consistent with the other changes. The approach you chose also LGTM.